### PR TITLE
Updates default request library

### PR DESCRIPTION
**Pull request by ajyey**
This PR updates `DEFAULT_REQUEST_LIBRARY` to `reqwest` to address #37 and also sets the default request library version

### CODE REVIEW

**Comment by AndyTitu** [comment]
Hey @ajyey, thank you for your contribution. Would you mind rebasing this PR on the latest `main`? We recently fixed an issue that caused the GitHub action workflows to not run on forked PRs, but on the latest `main` they should now exist.

File: src/onepassword/defaults.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 SDK_VERSION = "0010001"  # v0.1.0
 DEFAULT_INTEGRATION_NAME = "Unknown"
 DEFAULT_INTEGRATION_VERSION = "Unknown"
-DEFAULT_REQUEST_LIBRARY = "net/http"
+DEFAULT_REQUEST_LIBRARY = "reqwest"
```
Comment (AndyTitu) [inline src/onepassword/defaults.py:7]: We should also have a REQUEST_LIBRARY_VERSION default. Currently the one we're using in the closed source rust code has version `0.11.24` so we need to set it to that in order to get the internal logs fully accurate for debugging purposes.

Comment (AndyTitu) [reply src/onepassword/defaults.py:7]: Also, a little down in this same file we need to make the modification on:

```
"requestLibraryVersion": platform.python_version(),
```

to 

```
"requestLibraryVersion": REQUEST_LIBRARY_VERSION,
```

Comment (ajyey) [reply src/onepassword/defaults.py:7]: Done 👍

**Comment by ajyey** [comment]
> Hey @ajyey, thank you for your contribution. Would you mind rebasing this PR on the latest `main`? We recently fixed an issue that caused the GitHub action workflows to not run on forked PRs, but on the latest `main` they should now exist.

No problem. Done 👍

**Comment by AndyTitu** [comment]
Hey @ajyey, apologies for the delay here. Could you please rebase onto the latest main so we can try and make your pipeline pass. Thanks!

**Comment by AndyTitu** [comment]
The integration tests were successful for this forked branch as it can be noted from the last check of this PR! Merging now.

**Review by AndyTitu — APPROVED** [review]

**Pull request merged**